compile/field.go

```diff
@@ -85,9 +85,13 @@ func (r fieldRequiredness) isRequired(src *ast.Field) (bool, error) {
 //
 // disallowDefaultValue specifies whether the field is allowed to have a default
 // value.
+//
+// allowNegativeIDs controls whether negative field identifiers are allowed.
+// This also enables auto-assigning (negative) values for unset field IDs.
 type fieldOptions struct {
 	requiredness         fieldRequiredness
 	disallowDefaultValue bool
+	allowNegativeIDs     bool
```

Comment:
We'll need to expose the public Option for this inside [compile/options.go](https://github.com/thriftrw/thriftrw-go/blob/dev/compile/options.go), which will set it on the `*compiler`, and that'll plumb it to fieldOptions for compileStruct.

Reply:
Would you like these to be part of the existing `NonStrict` option's behavior, or as a new (to be named) option?

Reply:
The existing NonStrict flag should do. Thanks!

(Apologies for missing this yesterday.)